File: LeetCode/Python/Easy/1496-Path-Crossing/solution.py
class Solution:
    def isPathCrossing(self, path: str) -> bool:
        visited=set()

        visited.add((0, 0))

        current_x=0

        current_y=0

        for direction in path:
            if direction=='N':
                current_x-=1
            elif direction=='S':
                current_x+=1
            elif direction=='E':
                current_y+=1
            else:
                current_y-=1
            
            next_point=(current_x, current_y)

            if next_point in visited:
                return True
            
            visited.add(next_point)

        return False

File: LeetCode/Rust/Easy/1496-Path-Crossing/solution.rs
use std::collections::HashSet;

impl Solution
{
    pub fn is_path_crossing(path: String) -> bool
    {
        let mut visited: HashSet<(i32, i32)>=HashSet::new();

        visited.insert((0, 0));

        let mut current_x: i32=0;

        let mut current_y: i32=0;

        for direction in path.chars()
        {
            match direction
            {
                'N' =>
                {
                    current_x-=1;
                }
                'S' =>
                {
                    current_x+=1;
                }
                'E' =>
                {
                    current_y+=1;
                }
                'W' =>
                {
                    current_y-=1;
                }
                _ =>
                {
                    ()
                }
            }

            let next_point: (i32, i32)=(current_x, current_y);

            if visited.contains(&next_point)
            {
                return true;
            }

            visited.insert(next_point);
        }

        false
    }
}